base/linalg/bunchkaufman.jl

```diff
@@ -22,36 +22,38 @@ BunchKaufman(A::AbstractMatrix{T}, ipiv::Vector{BlasInt}, uplo::Char, symmetric:
 `bkfact!` is the same as [`bkfact`](@ref), but saves space by overwriting the
 input `A`, instead of creating a copy.
 """
-function bkfact!(A::StridedMatrix{<:BlasReal}, uplo::Symbol = :U,
-    symmetric::Bool = issymmetric(A), rook::Bool = false)
-
-    if !symmetric
-        throw(ArgumentError("Bunch-Kaufman decomposition is only valid for symmetric matrices"))
+function bkfact!(A::HermOrSym{T,S} where {T<:BlasReal,S<:StridedMatrix{T}}, rook::Bool = false)
+    if rook
+        LD, ipiv, info = LAPACK.sytrf_rook!(A.uplo, A.data)
+    else
+        LD, ipiv, info = LAPACK.sytrf!(A.uplo, A.data)
     end
+    BunchKaufman(LD, ipiv, A.uplo, true, rook, info)
+end
+function bkfact!(A::Symmetric{T,S} where {T<:BlasComplex,S<:StridedMatrix{T}}, rook::Bool = false)
     if rook
-        LD, ipiv, info = LAPACK.sytrf_rook!(char_uplo(uplo), A)
+        LD, ipiv, info = LAPACK.sytrf_rook!(A.uplo, A.data)
     else
-        LD, ipiv, info = LAPACK.sytrf!(char_uplo(uplo), A)
+        LD, ipiv, info = LAPACK.sytrf!(A.uplo, A.data)
     end
-    BunchKaufman(LD, ipiv, char_uplo(uplo), symmetric, rook, info)
+    BunchKaufman(LD, ipiv, A.uplo, true, rook, info)
 end
-function bkfact!(A::StridedMatrix{<:BlasComplex}, uplo::Symbol=:U,
-    symmetric::Bool=issymmetric(A), rook::Bool=false)
-
+function bkfact!(A::Hermitian{T,S} where {T<:BlasComplex,S<:StridedMatrix{T}}, rook::Bool = false)
     if rook
-        if symmetric
-            LD, ipiv, info = LAPACK.sytrf_rook!(char_uplo(uplo), A)
-        else
-            LD, ipiv, info = LAPACK.hetrf_rook!(char_uplo(uplo), A)
-        end
+        LD, ipiv, info = LAPACK.hetrf_rook!(A.uplo, A.data)
     else
-        if symmetric
-            LD, ipiv, info = LAPACK.sytrf!(char_uplo(uplo),  A)
-        else
-            LD, ipiv, info = LAPACK.hetrf!(char_uplo(uplo), A)
-        end
+        LD, ipiv, info = LAPACK.hetrf!(A.uplo, A.data)
+    end
```

Comment:
Likewise here, if you feel like condensing the five lines into one :). (`LD, ipiv, info = rook ? LAPACK.hetrf_rook!(A.uplo, A.data) : LAPACK.hetrf!(A.uplo, A.data)`.)